storage/block.go

```diff
@@ -43,7 +44,8 @@ type dbBlock struct {
 	opts     m3db.DatabaseOptions
 	start    time.Time
 	encoder  m3db.Encoder
-	stream   m3db.SegmentReader
+	segment  *m3db.Segment
+	srAlloc  m3db.SegmentReaderAllocate
```

Comment:
Would be in favor of just calling this `segmentReaderAlloc`, seems more readable and there's no super long lines of code as a side effect as its only really used in one place?